### PR TITLE
Update windows auth docs re: NTLM on Kestrel

### PR DESCRIPTION
**Pull request by anurse**
Docs update suggested in https://github.com/aspnet/AspNetCore/issues/14951

Need @Tratcher to confirm that I've interpreted the thread right :).

### CODE REVIEW

File: aspnetcore/security/authentication/windowsauth.md [hunk 1]
```diff
@@ -145,7 +145,7 @@ Use **either** of the following approaches:
 
 ## Kestrel
 
-The [Microsoft.AspNetCore.Authentication.Negotiate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Negotiate) NuGet package can be used with [Kestrel](xref:fundamentals/servers/kestrel) to support Windows Authentication using Negotiate, Kerberos, and NTLM on Windows, Linux, and macOS.
+The [Microsoft.AspNetCore.Authentication.Negotiate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Negotiate) NuGet package can be used with [Kestrel](xref:fundamentals/servers/kestrel) to support Windows Authentication using Negotiate and Kerberos on Windows, Linux, and macOS. Authentication using NTLM is supported on Windows only.
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]: The point in aspnet/AspNetCore#14951 was that NTLM was not directly supported on any platform, only via negotiate.

**Comment by anurse** [comment]
Update: Per @Tratcher 's comments and brief offline discussion, I just dropped NTLM from the doc entirely.

**Comment by guardrex** [comment]
:shipit::shipit::shipit::shipit::shipit::shipit::shipit:

**Comment by tornie2** [comment]
As I understand Kestrel could actually authenticate NTLM
But there is a problem with sending the challange / response

Would it be possible to build our own with middleware like this:

`public class ValidateAuthentication : IMiddleware
{
    public async Task InvokeAsync(HttpContext context, RequestDelegate next)
    {
        if (context.Request.Method == "OPTIONS")
        {
            await next(context);
            return;
        }

        if (context.User.Identity != null && context.User.Identity.IsAuthenticated)
        {
            await next(context);
        }
        else
        {
            context.Response.StatusCode = 401;
            context.Response.Headers["Proxy-Authenticate"] = "NTLM";
            context.Response.Headers["WWW-Authenticate"] = "NTLM";
            context.Response.Headers["Access-Control-Allow-Origin"] = context.Request.Headers["Origin"];
        }
    }
}`

**Comment by guardrex** [comment]
@tornie2 ... We don't work closed issues/PR. Please open a new issue using the **This page** feedback button and form at the bottom of the **English-US** topic. Use of the **This page** feedback form adds metadata to your GitHub issue that cross-links the topic and automatically pings the author.

Capture

**Comment by Tratcher** [comment]
@tornie2 That would not be enough. The implementation only looks for auth handshakes that start with "Negotiate". See
https://github.com/dotnet/aspnetcore/blob/d40eea990edd1ac7205a983cbe4d13a71fd4ad68/src/Security/Authentication/Negotiate/src/NegotiateHandler.cs#L101

**Comment by tornie2** [comment]
@Tratcher  Thanks for the tip. First tests indicate, it could actually work :-)

**Pull request merged**